tests/rptest/tests/wasm_partition_movement_test.py

```diff
@@ -0,0 +1,199 @@
+            num_nodes=1,
+            redpanda=self.redpanda,
+            topic=materialized_topic,
+            group_id="test_group",
+            on_record_consumed=self._on_data_consumed)
+        self.mconsumer.start()
+
+    def _await_consumer(self, limit, timeout):
+        wait_until(
+            lambda: self.mconsumer.total_consumed() >= limit,
+            timeout_sec=timeout,
+            err_msg=
+            "Timeout of after %ds while awaiting delivery of %d materialized records, recieved: %d"
+            % (timeout, limit, self.mconsumer.total_consumed()))
+        self.mconsumer.stop()
+
+    def _prime_env(self):
+        output_topic = "identity_output2"
+        self.start_redpanda_nodes(3)
+        spec = TopicSpec(name="topic2",
+                         partition_count=3,
+                         replication_factor=3)
+        self.client().create_topic(spec)
+        self._deploy_identity_copro([spec.name], [output_topic])
+        self.topic = spec.name
+        self.start_producer(num_nodes=1, throughput=10000)
+        self.start_consumer(1)
+        self.await_startup(min_records=500)
+        materialized_topic = construct_materialized_topic(
+            spec.name, output_topic)
+
+        def topic_created():
+            metadata = self.client().describe_topics()
+            self.logger.info(f"metadata: {metadata}")
+            return any([x['topic'] == materialized_topic for x in metadata])
+
+        wait_until(topic_created, timeout_sec=30, backoff_sec=2)
+
+        self._start_mconsumer(materialized_topic)
+        t, p = self._grab_input(spec.name)
+        return {
+            'topic': t,
+            'partition': p,
+            'materialized_topic': materialized_topic
+        }
+
+    @cluster(num_nodes=6)
+    def test_dynamic(self):
+        """
+        Move partitions with active consumer / producer
+        """
+        s = self._prime_env()
+        for _ in range(5):
+            _, partition, assignments = self._do_move_and_verify(
+                s['topic'], s['partition'])
+            self._verify_materialized_assignments(s['materialized_topic'],
+                                                  partition, assignments)
+
+        # Vaidate input
+        self.run_validation(min_records=500,
+                            enable_idempotence=False,
+                            consumer_timeout_sec=90)
+
+        # Wait for all output
+        self._await_consumer(self.consumer.total_consumed(), 90)
+
+        # Validate number of records & their content
+        self.logger.info(
+            f'A: {self.mconsumer.total_consumed()} B: {self.consumer.total_consumed()}'
+        )
+        assert self.mconsumer.total_consumed() == self.consumer.total_consumed(
+        )
+        # Since the identity copro was deployed, contents of logs should be identical
+        assert set(self.records_consumed) == set(self.result_data)
+
+    @ignore  # https://github.com/vectorizedio/redpanda/pull/3558
```

Comment:
This has passed for me locally with `--repeat 10`, marked as @ignore as this depends on changes from #3558

Reply:
@ignore directive removed now that #3558 is in